### PR TITLE
Rewire the Sticker button to be an Emoji Picker

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/1107

Things left to do:
- [x] Figure out a nice way to expose stickers (likely, using a new icon) in the composer

image

### CODE REVIEW

**Comment by fti7** [comment]
Any Updates here? :smile: Would be create to have this merged!

**Comment by t3chguy** [comment]
Blocked on designer who is busy doing FOSDEM graphics and Cross Signing stuff :( we have to be patient

**Comment by jacotec** [comment]
Any updates here? The current emoji solution is a real mess and I'd also like to see that PR merged sooner than later. I quite often write messages in my iPhone instead in my opened Riot-Web just because I know I need some emojis ...

**Comment by fti7** [comment]
Soon its March 2020.... any update from the Fosdem Designer? @lampholder please priotize such important UI stuff 😞

**Comment by nadonomy** [comment]
FYI all we've kicked off the design process on a new more extensible and scalable composer design for web, iOS & Android. We've opted for this as it means we can get things right in the long term.

Once we make good headway there, we'll run the design proposals through the major stakeholders per platform to figure out an execution plan, which will include very low hanging near-term tweaks (like this one!) that are aligned with long term direction.

**Comment by fti7** [comment]
@nadonomy do you have an rough estimation? Another 6 Months? :disappointed:

**Comment by nadonomy** [comment]
> @nadonomy do you have an rough estimation? Another 6 Months? 😞

Snark aside— unfortunately the designer working on this has been ill for the last 2 weeks (since the day after they started on the designs). Totally undesirable delay, but the designs will be ready when they're ready.

**Comment by t3chguy** [comment]
image

Latest ^

**Comment by aaronraimist** [comment]
Looks pretty good! Any chance https://github.com/vector-im/riot-web/issues/6998 could get fixed here as well? The composer is starting to get lots of buttons.

**Comment by aaronraimist** [comment]
Also in the tooltip "Show Stickers" has "stickers" capitalized but all of the other buttons there only have the first word capitalized

**Comment by t3chguy** [comment]
> Looks pretty good! Any chance vector-im/riot-web#6998 could get fixed here as well? The composer is starting to get lots of buttons.

Would love to, but that would mean waiting on product decision which would mean it'd miss next release so no.

> Also in the tooltip "Show Stickers" has "stickers" capitalized but all of the other buttons there only have the first word capitalized

Fair but not a regression so don't want to fix it

File: src/components/views/rooms/MessageComposer.js [hunk 1]
```diff
@@ -103,6 +104,32 @@ HangupButton.propTypes = {
...
+        contextMenu = <ContextMenu {...aboveLeftOf(buttonRect)} onFinished={closeMenu} catchTab={false}>
+            <EmojiPicker onChoose={addEmoji} showQuickReactions={true} />
+        </ContextMenu>;
+    }
+
+    return <React.Fragment>
+        <ContextMenuButton className="mx_MessageComposer_button mx_MessageComposer_emoji"
+                           onClick={openMenu}
+                           isExpanded={menuDisplayed}
+                           label={_t('Upload file')}
```
Comment (jryans) [inline src/components/views/rooms/MessageComposer.js:123]: Label seems not quite right...?

File: src/components/views/rooms/Stickerpicker.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ const STICKERPICKER_Z_INDEX = 3500;
 // Key to store the widget's AppTile under in PersistedElement
 const PERSISTED_ELEMENT_KEY = "stickerPicker";
 
+// TODO figure out where to expose it now that the EmojiPicker has taken its place
```
Comment (jryans) [inline src/components/views/rooms/Stickerpicker.js:38]: Hmm, so is it just going to be inaccessible for a while?

Comment (t3chguy) [reply src/components/views/rooms/Stickerpicker.js:38]: this comment is just stale, we now have a button for each

**Comment by telmich** [comment]
Ping @verysanghee - this is interesting for you

**Pull request merged**